### ISSUE TITLE
Permit underscores in method names without disabling INTL0003 entirely

### ISSUE THREAD

**Issue by Steborino** [IntelliTect/CodingGuidelines#105]
> DO NOT: Use underscores to break up words in an identifier. **Test methods**, UI generated event handlers and references to Resource files **are the exception**.

For testing methods, I need to suppress `INTL0003` as it checks for both PascalCasing and a lack of underscores. If the underscores check was extracted into a separate analyzer, I could suppress underscores without killing the PascalCasing check wholesale.

**Comment by Keboo** [comment]
Are the underscores used in unit test methods?

**Comment by Steborino** [comment]
Yes, the names of the methods — e.g. in the format `TestedMethodName_Scenario_ExpectedBehavior`.

**Comment by Keboo** [comment]
I agree with the proposal. @MarkMichaelis any objection to separating these two rules so they can be individually turned off?

**Comment by MarkMichaelis** [comment]
Great suggestion... I agree completely.

As a refinement, (in addition) we should not trigger the rule for test methods anyway.

**Comment by MarkMichaelis** [comment]
Do we know how to "recognize" a test method?  Would we need a hardcoded list of attributes (for each unit testing framework), or is there a known better way.

**Comment by Keboo** [comment]
As per discussion we should do the following:

- Split the analyzer into two analyzers. One for casing, one for underscores.
- The underscore analyzer should attempt to look at attributes on methods to determine if it is  unit test method and not flag if so. This is not expected to be perfect but cover the most common use cases.
